src/types.ts

```diff
@@ -2063,3 +2063,62 @@ export type User<UserType = UnknownType> = UserType & {
   teams?: string[];
   username?: string;
 };
+
+export type SegmentData = {
+  description: string;
+  filter: {
+    channel?: object;
+    user?: object;
```

Comment:
are these enriched objects from API?

Reply:
no, they're either a User or Channel filter (like queryusers/querychannels), example:

`{"id": 1223, "username": {$eq: "amin"}}`

Reply:
in that case let's pull @nhannah to help us with this type